controllers/src/main/java/io/syndesis/controllers/integration/online/DeleteHandler.java

```diff
@@ -33,27 +36,34 @@
     }
 
     @Override
-    public Set<Integration.Status> getTriggerStatuses() {
-        return Collections.singleton(Integration.Status.Deleted);
+    public Set<IntegrationState> getTriggerStatuses() {
+        return Collections.singleton(IntegrationState.Undeployed);
     }
 
     @Override
-    public StatusUpdate execute(Integration integration) {
+    public StatusUpdate execute(Integration integration, IntegrationRevision revision) {
+        //It's possible that we delete a draft version that never got deployed. In this case the version will be 0.
+        Integer version = revision.getVersion().orElse(0);
+
+        if (version == 0) {
+            return new StatusUpdate(0, IntegrationState.Undeployed);
+        }
+
         String token = integration.getToken().get();
         Tokens.setAuthenticationToken(token);
 
         OpenShiftDeployment deployment = OpenShiftDeployment
             .builder()
+            .revisionNumber(version)
             .name(integration.getName())
             .token(token)
             .build();
 
-        Integration.Status currentStatus = !openShiftService.exists(deployment)
+        IntegrationState currentStatus = !openShiftService.exists(deployment)
             || openShiftService.delete(deployment)
-            ? Integration.Status.Deleted
-            : Integration.Status.Pending;
+            ? IntegrationState.Undeployed
+            : IntegrationState.Pending;
```

Comment:
No "Pending" status, just leave either the currentStatus != targetStatus or set current status to "Error".